src/main/java/com/aws/greengrass/shadowmanager/sync/strategy/RealTimeSyncStrategy.java

```diff
@@ -0,0 +1,248 @@
+                    SyncRequest failedRequest = request;
+
+                    // tell queue this is not a new value so it merges correctly with any update that came in
+                    request = syncQueue.offerAndTake(request, false);
+
+                    // if queue was empty, we are going to immediately retrying the same request. For this case don't
+                    // use the default retry configuration - keep from spamming too quickly
+                    if (request == failedRequest) {
+                        retryConfig = FAILED_RETRY_CONFIG;
+                    }
+                } catch (InterruptedException e) {
+                    logger.atWarn(SYNC_EVENT_TYPE).log("Interrupted while waiting for sync requests");
+                    Thread.currentThread().interrupt();
+                } catch (ConflictException | ConflictError e) {
+                    logger.atWarn(SYNC_EVENT_TYPE)
+                            .cause(e)
+                            .addKeyValue(LOG_THING_NAME_KEY, request.getThingName())
+                            .addKeyValue(LOG_SHADOW_NAME_KEY, request.getShadowName())
+                            .log("Received conflict when processing request. Retrying as a full sync");
+                    // add back to queue to merge over any shadow request that came in while it was executing
+                    request = syncQueue.offerAndTake(new FullShadowSyncRequest(request.getThingName(),
+                            request.getShadowName()), true);
+                } catch (UnknownShadowException e) {
+                    logger.atWarn(SYNC_EVENT_TYPE)
+                            .cause(e)
+                            .addKeyValue(LOG_THING_NAME_KEY, request.getThingName())
+                            .addKeyValue(LOG_SHADOW_NAME_KEY, request.getShadowName())
+                            .log("Received unknown shadow when processing request. Retrying as a full sync");
+                    // add back to queue to merge over any shadow request that came in while it was executing
+                    request = syncQueue.offerAndTake(new FullShadowSyncRequest(request.getThingName(),
+                            request.getShadowName()), true);
+                } catch (Exception e) {
+                    logger.atError(SYNC_EVENT_TYPE)
+                            .cause(e)
+                            .addKeyValue(LOG_THING_NAME_KEY, request.getThingName())
+                            .addKeyValue(LOG_SHADOW_NAME_KEY, request.getShadowName())
+                            .log("Skipping sync request");
+                    request = syncQueue.take();
+                }
+            } while (!Thread.currentThread().isInterrupted());
+        } catch (InterruptedException e) {
+            logger.atWarn(SYNC_EVENT_TYPE).log("Interrupted while waiting for sync requests");
+            Thread.currentThread().interrupt();
```

Comment:
not sure that this is strictly necessary since the thread is going to exit anyway after catching the exception

Reply:
Right. I didn't change this function. I just refactored it from `SyncHandler`. Removed it now.